plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -198,40 +193,46 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sqlQuery += queryAddon
 
 		if p.Query[i].Version <= dbVersion {
-			rows, err := p.DB.Query(sqlQuery)
-			if err != nil {
-				p.Log.Error(err.Error())
-				continue
-			}
+			p.gatherMetricsFromQuery(acc, sqlQuery, p.Query[i].Tagvalue, p.Query[i].Timestamp, measName)
+		}
+	}
+	return nil
+}
 
-			defer rows.Close()
+func (p *Postgresql) gatherMetricsFromQuery(acc telegraf.Accumulator, sqlQuery string, tagValue string, timestamp string, measName string) {
+	var columns []string
 
-			// grab the column information from the result
-			if columns, err = rows.Columns(); err != nil {
-				p.Log.Error(err.Error())
-				continue
-			}
+	rows, err := p.DB.Query(sqlQuery)
+	if err != nil {
+		acc.AddError(err)
+		return
```

Comment:
Before this was only logged right? Are we going to introduce failures if people were ignoring this type of error before?

Reply:
Yes. I've changed it to `AddError` because @srebhan asked me about it here: https://github.com/influxdata/telegraf/pull/10066#discussion_r743856363

But `AddError` does pretty much the same:
```
// AddError passes a runtime error to the accumulator.
// The error will be tagged with the plugin name and written to the log.
func (ac *accumulator) AddError(err error) {
	if err == nil {
		return
	}
	ac.maker.Log().Errorf("Error in plugin: %v", err)
}
```